### DISCUSSION TITLE
Ability to terminate a running CVC5 instance

### DISCUSSION THREAD

**Discussion by baierd** [cvc5/cvc5#8003] [Q&A] [answered]
Hello,
i use the Java bindings of CVC5 in a [project](https://github.com/sosy-lab/java-smt) and noticed that there seems to be no interruption or termination callback (or i missed it in the Java API). Would it be possible to add this feature to the CVC5 bindings?

Thank you for your work!
Best,
Daniel

**Comment by nafur** [comment, marked as answer]
Hi Daniel,
~~right now we don't expose such a callback in any of our APIs. However, you can inspect `cvc5.api.Result` and figure out whether an unknown result was due to a timeout, resource out or alike.~~
I first understood you were asking for a callback to know **when** the solver was interrupting itself.

The actual answer: no, there is no explicit API method to interrupt the currently running solver. However, you can impose certain limits before starting it (`rlimit`, `tlimit` etc). You can probably abuse them to do what you want as well: calling `setOption("tlimit-per", "1")` asynchronously would probably just work and trigger a self-induced interrupt (almost) immediately.
